sdk/storage/AzureStorageBlob/Source/Transport/URLSessionTransferManager.swift

```diff
@@ -446,6 +492,32 @@ internal final class URLSessionTransferManager: NSObject, TransferManager, URLSe
         self.transfer(transfer, didUpdateWithState: transfer.state, andProgress: nil)
     }
 
+    // MARK: Misc Methods
+
+    func handleNetworkTransition() {
+        let status = networkStatus.publicValue
+
+        let clientEnumerator = clients.objectEnumerator()
+        while let client: StorageBlobClient = clientEnumerator?.nextObject() as? StorageBlobClient {
+            let downloadPolicy = client.options.downloadNetworkPolicy
+            if downloadPolicy.shouldTransfer(withStatus: status) {
+                if downloadPolicy.enableAutoResume.contains(status!) {
```

Comment:
We should be able to avoid these force-unwraps right? Something like the following at the beginning of the while-loop?

```swift
guard let status = status else {
    client.transfers.pauseAll()
    continue
}
```

Reply:
Done